src/main/java/io/nats/client/api/OrderedConsumerConfig.java

```diff
@@ -0,0 +1,169 @@
+
+package io.nats.client.api;
+
+import java.time.Duration;
+import java.time.ZonedDateTime;
+
+import static io.nats.client.support.MiscUtils.generateConsumerName;
+import static io.nats.client.support.Validator.emptyOrNullAs;
+
+public class OrderedConsumerConfig {
+    private final ConsumerConfiguration cc;
+
+    private OrderedConsumerConfig(ConsumerConfiguration cc) {
+        this.cc = cc;
+    }
+
+    /**
+     * Macro to start a OrderedConsumerConfig builder
+     * @return builder
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * OrderedConsumerConfig can be created using a Builder.
+     * The builder supports chaining and will create a default set of options if
+     * no methods are calls, including setting the filter subject to "&gt;"
+     */
+    public static class Builder {
+        final ConsumerConfiguration.Builder builder;
+
+        public Builder() {
+            builder = new ConsumerConfiguration.Builder()
+                .name(generateConsumerName())
+                .filterSubject(">");
+        }
+
+        /**
+         * Sets the delivery policy of the OrderedConsumerConfig.
+         *
+         * @param policy the delivery policy.
+         * @return Builder
+         */
+        public Builder deliverPolicy(DeliverPolicy policy) {
+            builder.deliverPolicy(policy);
+            return this;
+        }
+
+        /**
+         * Sets the start sequence of the OrderedConsumerConfig.
+         *
+         * @param sequence the start sequence
+         * @return Builder
+         */
+        public Builder startSequence(long sequence) {
+            builder.startSequence(sequence);
+            return this;
+        }
+
+        /**
+         * Sets the start time of the OrderedConsumerConfig.
+         * @param startTime the start time
+         * @return Builder
+         */
+        public Builder startTime(ZonedDateTime startTime) {
+            builder.startTime(startTime);
+            return this;
+        }
+
+        /**
+         * Sets the filter subject of the OrderedConsumerConfig.
+         *
+         * @param filterSubject the filter subject
+         * @return Builder
+         */
+        public Builder filterSubject(String filterSubject) {
+            builder.filterSubject(emptyOrNullAs(filterSubject, ">"));
+            return this;
+        }
+
+        /**
+         * Sets the replay policy of the OrderedConsumerConfig.
+         *
+         * @param policy the replay policy.
+         * @return Builder
+         */
+        public Builder replayPolicy(ReplayPolicy policy) {
+            builder.replayPolicy(policy);
+            return this;
+        }
+
+        /**
+         * sets the max amount of expire time for the server to allow on pull requests.
+         *
+         * @param maxExpires the max expire duration
+         * @return Builder
+         */
+        public Builder maxExpires(Duration maxExpires) {
```

Comment:
I am wondering whether it makes sense to allow setting `maxExpires`, `maxBytes` and `maxBatch` for ordered consumers. After all, those values will be set in consume configuration and ordered consumer is not meant to be used outside of the application. So does it make sense to set limits on itself when it's always the same process that will use it?

Reply:
@piotrpio Not sure. All the other simplification allows them, they are limits. The user still can set batch size messages or bytes. I suppose I could interrogate that before I make the final configuration.

Reply:
Yes, user can set batch sizes etc on `Consume`, I was just wondering what's the point of setting those limits if only your application will be setting the values in `Consume`. It certainly does not hurt to give those options, I just don't see the reason to ever use them.

Reply:
I guess this needs discussion / adr?

Reply:
why not just let it be what the server defaults?

Reply:
Ordered consumer should have the min set of options possible.